activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1055,7 +1055,13 @@ def arel_columns(columns)
           case field
           when Symbol
             field = field.to_s
-            arel_column(field) { connection.quote_table_name(field) }
+            arel_column(field) do
+              table, sep, column = field.to_s.rpartition(".")
```

Comment:
This only accounts for a single dot, but you can have multiple dots in a table name, eg. `field = :"a.b.c"`

Reply:
This uses `rpartition`, so it splits from the last dot.
For your example, the result would be `['a.b', '.', 'c']`

Reply:
Can you fix it in the `quote_table_name`?
Here is not only place we use `quote_table_name`.

Reply:
hm.. I see the problem.

https://github.com/rails/rails/blob/825d70b4069df3411b77636c98ca668292bbc03a/activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb#L42-L44

We should use `quote_table_name` only to quote actual table name or schema qualified table name.
Here is one of the case of quoting table name qualified *column name*.
We probably need new method for the purpose.

Reply:
I can transfer my logic to `Quoting` under the name `quote_table_and_column_name`.

The issue is how to deal with column names which might also contain dots. My fix doesn't deal with those, that's why I didn't want to make a method; it wasn't as general purpose as I would like.

Reply:
At least the `quote_table_name` is also used for quoting the order by column.
4650874073bef4f02da3c2b0bfc4ba2a6fd5d562 would be broken by this change for now.

Reply:
Alright, thnx, I'll rebase and fix+test again when I move the logic to Quoting.